### ISSUE TITLE
fs.stat() returns [native code] instead of a Stats object

### ISSUE THREAD

**Issue by hippietrail** [oven-sh/bun#624]
### Version

0.1.3

### Platform

Darwin Andrews-M1-Air 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:29 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T8101 arm64

### What steps will reproduce the bug?

Run this code:
```js
const fs = require("fs")

fs.stat(".", (e, s) => console.log(e, s))
```

### How often does it reproduce? Is there a required condition?

every time. 100%

### What is the expected behavior?

```
null Stats {
  dev: 16777233,
  mode: 16877,
  nlink: 12,
  uid: 501,
  gid: 20,
  rdev: 0,
  blksize: 4096,
  ino: 35750497,
  size: 384,
  blocks: 0,
  atimeMs: 1657628302382.1816,
  mtimeMs: 1657628302350.719,
  ctimeMs: 1657628302350.719,
  birthtimeMs: 1657178031769.6665,
  atime: 2022-07-12T12:18:22.382Z,
  mtime: 2022-07-12T12:18:22.351Z,
  ctime: 2022-07-12T12:18:22.351Z,
  birthtime: 2022-07-07T07:13:51.770Z
}
```

### What do you see instead?

null [native code]

**Comment by Jarred-Sumner** [comment]
I think this is a `console.log` bug and not a bug with the acutal stats object

If you do `Object.keys(statObject)`, does it return keys?

**Comment by hippietrail** [comment]
Yes `Object.keys(statObject)` does return the keys.
I don't think console.log is the only problem though because I found the problem in more complicated code where I use `Object.assign()` to combine a `Dirent` and a `Stats`.

I'll play a bit to find another minimal example.

**Comment by hippietrail** [comment]
```js
    console.log(Object.assign({}, s))
    console.log(Object.assign(s, {}))
```
outputs
```
{ }
[native code]
```

**Comment by Jarred-Sumner** [comment]
I suspect the problem is that bun is using the JSC C API for this type and the C API generally just "feels weird" compared to regular objects. The objects created semantically don't behave the same as regular JS objects, as well as being slower to get properties. 

Bun should switch to JSC C++ API for this type and the problems will be go away. It can use `JSC::LazyClassStructure`.

**Comment by hippietrail** [comment]
I just found out that `readdir()` has the same problem as `stat()`

But it turns out they do return objects, because you can access the properties of those objects. But if you use `typeof` on them, they return `function` and if you try to print them they return `[native code]`

**Comment by Electroid** [comment]
This is fixed as of `0.2.2`! Feel free to comment if you can reproduce it, and we can re-open the issue.
